src/ol/renderer/canvas/TileLayer.js

```diff
@@ -735,7 +769,7 @@ class CanvasTileLayerRenderer extends CanvasLayerRenderer {
             if (opt_tileCallback !== undefined) {
               opt_tileCallback(tile);
             }
-          } else {
+          } else if (tileVisible) {
             tileSource.useTile(z, x, y, projection);
```

Comment:
This is unrelated to the changes in this PR, but shouldn't the used tiles increase the `tileCount` when they are cached?